### DISCUSSION TITLE
add algorithm param to /point endpoint

### DISCUSSION THREAD

**Discussion by JinIgarashi** [developmentseed/titiler#807] [Feature Request]
`/point` endpoint can be very powerful and useful to get values by a specific coordinate

https://titiler.xyz/api.html#/Cloud%20Optimized%20GeoTIFF/point_cog_point__lon___lat__get

However, currently there is no query params to apply algorithm for it. Because of this, we are unable to check actual value for a dataset after applying an algorithm.

Is it possible to add algorithm param to the endpoint?

**Comment by vincentsarago** [comment]
the `Algorithm` base class was designed for ImageData

https://github.com/developmentseed/titiler/blob/393614149e47fd6e0711cb30cb67b24c4b9f5388/src/titiler/core/titiler/core/algorithm/base.py#L26-L29

this is all coming down to rio-tiler either using ImageData and PointData. Both classes assume the data array (numpy masked array) to be in a different shape 

https://github.com/cogeotiff/rio-tiler/blob/main/rio_tiler/models.py#L178-L182
https://github.com/cogeotiff/rio-tiler/blob/main/rio_tiler/models.py#L298-L299

so having algorithm which could handle both is tricky, which is why we don't have the option in the `/point` endpoint
